core_impl/src/main/java/io/opencensus/implcore/stats/CurrentStatsState.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2017, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.implcore.stats;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+import io.opencensus.stats.StatsCollectionState;
+import io.opencensus.stats.StatsComponent;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * The current {@link StatsCollectionState} for a {@link StatsComponent}.
+ *
+ * <p>This class allows different stats classes to share the state in a thread-safe way.
+ */
+@ThreadSafe
+public final class CurrentStatsState {
+  private volatile StatsCollectionState currentState = StatsCollectionState.ENABLED;
+  private volatile boolean isRead;
+
+  public StatsCollectionState get() {
+    return currentState;
+  }
+
+  void setRead() {
+    isRead = true;
+  }
+
+  void set(StatsCollectionState state) {
+    checkState(!isRead, "State was already read, cannot set state.");
+    currentState = checkNotNull(state, "state");
```

Comment:
If the goal is to ensure that getState always returns the same value, then I think this class should synchronize access to the two fields.  Otherwise, the state could be read between these two lines.